clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -100,10 +100,7 @@ public int throttleTimeMs() {
     @Override
     public Map<Errors, Integer> errorCounts() {
         Map<Errors, Integer> errorCounts = new HashMap<>();
-        data.topics().forEach(metadata -> {
-            metadata.partitions().forEach(p -> updateErrorCounts(errorCounts, Errors.forCode(p.errorCode())));
```

Comment:
According to https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/metadata/ZkMetadataCache.scala#L103, it is possible that partition-level has error but topic-level gets none. Hence, it produces inaccurate count of "error" if we remove this line.

If we want to make correct count of `error` and `none`, we have to add following check.

1. if topic has error, we don't need to loop partitions.
2. if topic has no error, we have to loop partitions.

@ijuma WDYT?